js/data.js

```diff
@@ -6,3 +6,15 @@ let data = {
   editing: null,
   nextEntryId: 1,
 };
+
+window.addEventListener('beforeunload', handleBeforeUnload);
+
+function handleBeforeUnload(e) {
+  const entryHistoryJSON = JSON.stringify(data);
+  localStorage.setItem('entries', entryHistoryJSON);
+}
+
+const previousEntriesJSON = localStorage.getItem('entries');
+if (previousEntriesJSON !== null) {
+  data = JSON.parse(localStorage.getItem('entries'));
```

Comment:
You already have a variable `previousEntriesJSON` on line 17 that does the `getItem`. You should just utilize that variable rather than doing this again